### ISSUE TITLE
Different performance when pre-training on A100 and H100

### ISSUE THREAD

**Issue by MingliangLiang3** [mlfoundations/open_clip#1028]
I pre-trained the model on A100 and H100, but I got different results. Here is my training code:

**Pre-training on A100, the performance is {"imagenet-zeroshot-val-top1": 0.16732, "imagenet-zeroshot-val-top5": 0.33264}**

torchrun --nproc_per_node=4 --master_port=25633 open_clip_train/main.py \
  --save-frequency=1 \
  --report-to=tensorboard \
  --train-data="../data/cc3m/cc3m_train/cc3m-train-{00000..00192}.tar" \
  --train-num-samples=1922750 \
  --imagenet-val="../data/imagenet_validation/val" \
  --csv-img-key=image \
  --csv-caption-key=caption \
  --model=ViT-B-16 \
  --resume='latest' \
  **--batch-size=800 \**
  --aug-cfg scale='(0.50, 1.0)' \
  --force-patch-dropout=0.50 \
  --lr=1e-3 \
  --epochs=30 \
  --wd=0.2 \
  --workers=8 \
  --seed=42 \
  --local-loss \
  --gather-with-grad \
  --force-custom-text \
  **--precision=amp_bf16 \**
  --name cc3m/${pre_trained}

**Pre-training on A100, the performance is {"imagenet-zeroshot-val-top1": 0.1036, "imagenet-zeroshot-val-top5": 0.23072}**

torchrun --nproc_per_node=4 --master_port=25633 open_clip_train/main.py \
  --save-frequency=1 \
  --report-to=tensorboard \
  --train-data="../data/cc3m/cc3m_train/cc3m-train-{00000..00192}.tar" \
  --train-num-samples=1922750 \
  --imagenet-val="../data/imagenet_validation/val" \
  --csv-img-key=image \
  --csv-caption-key=caption \
  --model=ViT-B-16 \
  --resume='latest' \
  --batch-size=2200 \
  --aug-cfg scale='(0.50, 1.0)' \
  --force-patch-dropout=0.50 \
  --lr=1e-3 \
  --epochs=30 \
  --wd=0.2 \
  --workers=8 \
  --seed=42 \
  --local-loss \
  --gather-with-grad \
  --force-custom-text \
  **--precision=amp_bf16 \**
  --name cc3m/${pre_trained}

**I only change the batch-size, because the GPU memory is different.**

**Comment by rom1504** [comment]
That's expected, increasing batch size improves quality of models with clip
as you get more negative samples in the batch

**Comment by MingliangLiang3** [comment]
> That's expected, increasing batch size improves quality of models with clip
> as you get more negative samples in the batch
> […](#)

But when I pre-train the model on H100, I use a larger batch size and I get lower performance.

**Comment by rwightman** [comment]
something to consider, the `--warmup` param for LR warmup is in steps. At 2200 x 4, the defaul 10,000 steps is 88M samples, or more than your 30 epochs of training. At 800x4, it's 32M so you finish warmup a little after half way through train.

The default warmup steps was set for large datasets like 400M - 2B samples and fairly large batch sizes, it also works for modest datasets and small batch sizes, you need to reduce it for small datasets and larger batch sizes.
